contracts/mocks/HashMock.sol

```diff
@@ -0,0 +1,14 @@
+pragma solidity ^0.5.2;
+pragma experimental ABIEncoderV2;
+
+import {Hash} from "../Hash.slb";
+
+
+contract HashMock {
+    using Hash for uint[];
+
+    function testToHashArray(uint256[] memory array, bytes32 sibling) public pure returns (bytes32) {
+        return array.toHash(sibling);
```

Comment:
a question out of interest/ignorance:
why not integrate this into `EVMStack.slb` ?
the `fromArray(uint[] memory sArr)` function could take an additional `sibling` element.
the `toArray(Stack memory self)` function could return an array and sibling.

Reply:
I imagine all these hash will be put on `Hash.slb`.
Then in a contract `StateHash.sol` for example, we would have:
```
using Hash for EVMStack;
using Hash for EVMMemory;
...
```
It's personally feel better, because I know they have a hash function when I see that.
If putting `toHash` in `EVMStack`, I'd want those lib extense a `Hashable` or something.

Reply:
kind of decorator pattern, makes sense